R/req-as-curl.R

```diff
@@ -0,0 +1,224 @@
+#' Translate an httr2 request to a curl command
+#'
+#' Convert an httr2 request object to equivalent curl command line syntax.
+#' This is useful for debugging, sharing requests, or converting to other tools.
+#'
+#' @inheritParams req_perform
+#' @return A character string containing the curl command.
+#' @export
+#' @examples
+#' @seealso [curl_translate()]
+#' \dontrun{
+#' # Basic GET request
+#' request("https://httpbin.org/get") |>
+#'   req_as_curl()
+#'
+#' # POST with JSON body
+#' request("https://httpbin.org/post") |>
+#'   req_body_json(list(name = "value")) |>
+#'   req_as_curl()
+#'
+#' # POST with form data
+#' request("https://httpbin.org/post") |>
+#'   req_body_form(name = "value") |>
+#'   req_as_curl()
+#' }
+req_as_curl <- function(req) {
+  # validate the request
+  check_request(req)
+
+  # Extract URL
+  url <- req_get_url(req)
+
+  # use the request's method if it is set, otherwise infer
+  method <- req$method %||%
+    {
+      if (!is.null(req$body$data)) {
+        "POST"
+      } else {
+        "GET"
+      }
+    }
+
+  # we will append to cmd_args to build up the request
+  cmd_args <- c()
+
+  # if the method isn't GET, it needs to be specified with `-X`
+  if (method != "GET") {
+    cmd_args <- c(cmd_args, paste0("-X ", method))
+  }
+
+  # get headers and reveal obfuscated values
+  headers <- req_get_headers(req, redacted = "reveal")
+
+  # if headers are present, add them using -H flag
+  if (!rlang::is_empty(headers)) {
```

Comment:
Again, you don't need to first check if the vector is empty, if it is empty the for loop already doesn't do anything.